staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -92,7 +95,7 @@ func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig, scheme *r
 	if err := o.Authorization.ApplyTo(&config.Config.Authorization); err != nil {
 		return err
 	}
-	if err := o.Audit.ApplyTo(&config.Config); err != nil {
+	if err := o.Audit.ApplyTo(&config.Config, config.ClientConfig, config.SharedInformerFactory, o.ProcessInfo, nil, nil); err != nil {
```

Comment:
passing `nil, nil` looks suspicious

Reply:
this goes in to some previous conversations of how to get the auth and service resolver in here, I could add them to the recommend options as well if we think that makes sense

Reply:
ok updated with a `WebhookOptions` struct